### DISCUSSION TITLE
How to: Start Livestream from motion detection (H11 and Mini)

### DISCUSSION THREAD

**Discussion by dnewman-gpsw** [gopro/labs#381] [General]
This is the script pseudo code for what is needed to Livestream via a motion detection trigger.

```
  JoinWiFi();                //have the WiFi connection ready
  L=0;                       //initialize variable L for our Live streaming
loop:
  M=motion();                //read the current motion level
  if(L<1 and M>2.0)          //if not yet Live, and the motion is greater than 2%
     StartLive(), L=2, A=0;  //Start, set the live flag, reset the counter
  if(L>1 and M<1.0)          //if Live and the motion is low
     A=A+1;                  //start counting to stop the stream for lack of motion 
  else
     A=0;                    //else reset the counter
  if(A > 99)                 //if counter great than 100
     StopLive(), L=0, A=0;   //Stop, and set live flag to 0
  print(“vars $M $L $A”)     //print some debug.
  goto loop;                 //repeat
```

As a QR Code:
!W=L0=Mm<L1>M2!GL+=L2+=A0>L1<M1=A+1~=A0>A99!E+=L0+=A0oMPMSG="dbg $M $L $A"!R5

Or runs as a boot script:
!MBOOT="!Lboot"!SAVEboot=!W=L0=Mm<L1>M2!GL+=L2+=A0>L1<M1=A+1~=A0>A99!E+=L0+=A0oMPMSG="dbg $M $L $A"!R5

**Comment by chrisrscuba** [comment]
Question, what is the "!R5" command.  Does that mean Repeat but ignore the first 5 characters?  Is there a page where that is documented?  I think it could be very useful for one of my projects.  Thanks in advance!

**Reply by chrisrscuba** [reply]
Never mind, found it !timeRx  ..

**Comment by Fougl** [comment]
Hi, Is this already possible in the mix with contninuous mode? Im not sure how to live steram to NGINX rtmp server, how to use custom url here? Also, the above boot script starts the live stream without any motion... I am using firmware LABS_HERO11_02_01_70

Will motion detection work in tandem with live stream only with Hero 11?

**Reply by dnewman-gpsw** [reply]
I don't know what continuous mode is.

RTMP is still setup as shown here https://gopro.github.io/labs/control/rtmp/

Motion triggering of livestream only works with H11 and only on the current Labs firmware (V2.10.70.)

**Reply by Fougl** [reply]
Thanks for the quick reply as always. With continuous I mean repeat after every record.

RTMP is set once and then it uses that one every time? Will this feature work with H11 mini as well in the future?

**Reply by dnewman-gpsw** [reply]
!R the repeats the command.

!Mwxyz  <- all commands in the format and permanently saved. 

Mini should also Livestream.
